### PR TITLE
Adding application vulnerability rules

### PR DESCRIPTION
**Pull request by Moti-H**

### CODE REVIEW

**Comment by frack113** [comment]
HI,
Thanks you for your participation.

Some general comments: 
- No need to add `modified` for a new rule
- `critical` level is when you have no FP https://github.com/SigmaHQ/sigma-specification/blob/main/Sigma_specification.md#level
- [local_file_read.yml](https://github.com/SigmaHQ/sigma/pull/4034/files#diff-450c911251b669f1ae611da92334901ba8c198a04e82e7b830658270e0f7a2ca)  check a557ffe6-ac54-43d2-ae69-158027082350 (huawei_bgp_auth_failed.yml)
- `references` is for link (https://github.com/SigmaHQ/sigma-specification/blob/main/Sigma_specification.md#description-optional) . You can use description  to add more information.
- In `logsource` , `definition` is usefull to help to find the log eg in Node.js there are console.log, console.error and LOG_LEVEL... What should I look for if I have this product tomorrow at ?

**Comment by Moti-H** [comment]
Hey @frack113 
I fixed the rules based on your remarks, please let me know if there's anything else needed here

**Comment by nasbench** [comment]
@Moti-H can you please provide better references that site the logs you're using? This would be very helpful to anyone using the rules. To know what to expect. If you did the test yourself you can post log samples in the comments here.

Also, I don't think the rules (except for maybe e032f5bc-4563-4096-ae3b-064bab588685) can be of level high. As they are using error keywords that do not necessarily mean an attack. This means `medium` or `low` is best suited depending on how often these errors occur in prod.

For the definition and as @frack113 mentioned. Please don't use a generic one for all. Indicate what file or files are expected to have these kinds of logs. This would help the readers to know what files to collect.

**Comment by Moti-H** [comment]
Hey @nasbench

Regarding the log sources, it really depends on the application's deployment, as every app writes logs differently.
It is very common to have all application logs bundled together in a rotating log file (e.g 'invoices-service.14.02.log').
These logs normally include all log levels (INFO, ERROR, FATAL etc.) So it's hard to pinpoint a specific file.

Those keywords come from my personal experience with running 3K microservices, if you see the keywords in your application error logs it usually means trouble (either a vulnerability or an active exploitation of one), with a pretty high accuracy rate.
Seeing `process execution` & `template rendering` errors in production apps is a red flag.
That is why I think they should be ranked as high-risk.

I wrote a [blog post](https://www.wix.engineering/post/threat-and-vulnerability-hunting-with-application-server-error-logs) about it and I'm going to [talk about](https://dublin.globalappsec.org/schedule/) the topic at Global AppSec Dublin in a few days. I'd like to give reference to the rules and encourage more contributions from the AppSec community.

It's basically the same principle as the SQL Injection rule (8a670c6d-7189-4b1c-8017-a417ca84a086) - You shouldn't see lots of such errors in production apps.

Here for example I have a piece of code vulnerable to SpEL Injection (which leads to RCE):
```
    @PostMapping("/trigger-spel-injection")
    public String triggerSpelInjection(@RequestParam(value = "spelInput", defaultValue = "echo") String spelInput) {

        ExpressionParser parser = new SpelExpressionParser();
        Expression exp = parser.parseExpression(spelInput);
        String message = (String) exp.getValue();
        return String.format(message);
    }
```

An input that triggers the RCE would look like this:
`T(java.lang.Runtime).getRuntime().exec('someCommand')`

Let's say that for some reason the command execution failed (typo, the binary doesn't exists on the OS, no execute permissions etc.), it will trigger an error at the server with the following message:
```
Servlet.service() for servlet [dispatcherServlet] in context with path [] threw exception [Request processing failed: org.springframework.expression.ExpressionInvocationTargetException: A problem occurred when trying to execute method 'exec' on object of type [java.lang.Runtime]] with root cause
```

Followed by the stack trace:
```
java.io.IOException: error=2, No such file or directory
	at java.base/java.lang.ProcessImpl.forkAndExec(Native Method) ~[na:na]
	at java.base/java.lang.ProcessImpl.<init>(ProcessImpl.java:314) ~[na:na]
	at java.base/java.lang.ProcessImpl.start(ProcessImpl.java:244) ~[na:na]
	at java.base/java.lang.ProcessBuilder.start(ProcessBuilder.java:1110) ~[na:na]
	at java.base/java.lang.ProcessBuilder.start(ProcessBuilder.java:1073) ~[na:na]
	at java.base/java.lang.Runtime.exec(Runtime.java:594) ~[na:na]
	at java.base/java.lang.Runtime.exec(Runtime.java:418) ~[na:na]
	at java.base/java.lang.Runtime.exec(Runtime.java:315) ~[na:na]
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[na:na]
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77) ~[na:na]
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:na]
	at java.base/java.lang.reflect.Method.invoke(Method.java:568) ~[na:na]
	at org.springframework.expression.spel.support.ReflectiveMethodExecutor.execute
...
... truncated
```

The keywords in the error message and the stack trace are an indication of a vulnerability, and in some cases of active exploitation.

**Comment by nasbench** [comment]
@Moti-H Thanks for your input and detailed explanation :D much appreciated.

Regarding the ref If the slides or recording of your talk will be public, maybe we'll add them later for a more broader context. I will also add the blog you linked.

Regarding the severity, as you have much more experience we'll keep it as high for the reasons you described.

I will make some stylistic and wording suggestions for the titles. As I think they might be confusing for newcomers when reading a `JNDI Injection in the JVM` they think the injection is in the JVM itself for example. See if you like them or if you have some input maybe I'm wrong.

Thanks.

**Comment by frack113** [comment]
What do you think to change `product: jvm` to `product: java` ?

**Comment by Moti-H** [comment]
Yeah I get the confusion, maybe it's better to drop the " in JVM" part altogether.

I specified JVM and not only Java since these rules are also relevant for other JVM based runtimes (Scala, Kotlin etc.)
@nasbench would you like me to change anything else in the PR or are we good to go?

One last thing, if you do make changes - I accidentally slipped two keywords that are too broad and may generate false positives in rule `d65f37da-a26a-48f8-8159-3dde96680ad2`
```
        - java.lang.Runtime.exec
        - java.lang.UNIXProcess
```

This is because they can be part of a malicious payload, but they don't necessarily mean that there was a successful interaction with Java's `ProcessBuilder`
If it's not too much of a trouble, can you please remove them?

Thanks!

**Review by nasbench — APPROVED** [review]

**Comment by nasbench** [comment]
@Moti-H I made the changes you requested and updated the title wording to be a little bit more generic and suspicious (as it should be - so people would look at those alerts 😛 ). Check it and see if you have some final remarks maybe.

@frack113 could please give this a last double-check? It LGTM. See if you can find something from your side.

File: rules/application/jvm/java_jndi_injection.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+title: Potential JNDI Injection Exploitation In JVM Based Application
+id: bb0e9cec-d4da-46f5-997f-22efc59f3dca
+status: experimental
+description: Detects potential JNDI Injection exploitation. Often coupled with Log4Shell exploitation.
+references:
+    - https://www.shiftleft.io/blog/log4shell-jndi-injection-via-attackable-log4j/
```
Comment (frack113) [inline rules/application/jvm/java_jndi_injection.yml:6]:
```suggestion
```
get a 404

Comment (frack113) [reply rules/application/jvm/java_jndi_injection.yml:6]: find this https://secariolabs.com/research/analysing-and-reproducing-poc-for-log4j-2-15-0

File: rules/application/jvm/java_spel_injection.yml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    - https://owasp.org/www-community/vulnerabilities/Expression_Language_Injection
+    - https://www.wix.engineering/post/threat-and-vulnerability-hunting-with-application-server-error-logs
+author: Moti Harmats
+date: 2023/02/11
+tags:
+    - attack.initial_access
+    - attack.t1190
+logsource:
+    category: application
+    product: jvm
```
Comment (frack113) [inline rules/application/jvm/java_spel_injection.yml:15]: The rule is for Spring Framework so product should be springframework.

**Review by frack113 — APPROVED** [review]

**Pull request merged**